### ISSUE TITLE
Clang-tidy 'use-after-move' not detected in constructors

### ISSUE THREAD

**Issue by blitz-research** [llvm/llvm-project#56195]
Hi,

Clang-tidy is not correctly finding the 'use after move' error in the constructor here (and in all ctors AFAICT), although it is in the main code. This is with 'bugprone-use-after-move' enabled.

```cpp
class Test {
	std::string m_path;
	size_t m_size;

	explicit Test(std::string path)
		: m_path(std::move(path)), m_size(path.size()) { // Misses use-after-move of 'path' here...
		auto tmp = m_path + "!";
		test(std::move(tmp));
		std::cout << tmp.size(); // Correctly flags this though
	}

	void test(std::string str) {
		std::cout << str;
	}
};
```
I'm using the clang-tidy that's built-in to the last (2022.2) CLion and I'm not sure what version it is.

Bye!
Mark

**Comment by PiotrZSL** [comment]
Issue still exist on main.
Looks like initializer list is currently fully ignored.
I will try to fix this.

**Comment by PiotrZSL** [comment]
Same as #51844
